packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.ts

```diff
@@ -376,29 +386,31 @@ export const updateHTML = (props: UpdateHTMLProps): { updatedHTML: string; updat
             if (startIndex !== tag.plainTextBeginIndex && startIndex !== closingTagInfo.plainTextEndIndex) {
               // mention tag should be deleted when user tries to edit it in the middle
               result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex) + processedChange;
-              changeNewEndIndex = tag.plainTextBeginIndex + processedChange.length;
+              changeNewEndIndex = tag.plainTextBeginIndex + revertEscapedHtmlCharacters(processedChange).length;
```

Comment:
The property from[ this comment](https://github.com/Azure/communication-ui-library/pull/3226/files#r1234236489) can probably be used for `revertEscapedHtmlCharacters` call too